selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitPanel.java

```diff
@@ -0,0 +1,214 @@
+
+/** @author Mykola Morhun */
+public class GitPanel {
+  private final SeleniumWebDriver seleniumWebDriver;
+  private final ActionsFactory actionsFactory;
+  private final WebDriverWait uiWait;
+  private final WebDriverWait loadWait;
+
+  @Inject
+  public GitPanel(SeleniumWebDriver seleniumWebDriver, ActionsFactory actionsFactory) {
+    this.seleniumWebDriver = seleniumWebDriver;
+    this.actionsFactory = actionsFactory;
+
+    this.uiWait = new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
+    this.loadWait = new WebDriverWait(seleniumWebDriver, LOAD_PAGE_TIMEOUT_SEC);
+    PageFactory.initElements(seleniumWebDriver, this);
+  }
+
+  private interface Locators {
+    String GIT_PANEL_BUTTON_ID = "gwt-debug-partButton-Git";
+    String GIT_PANEL_ID = "gwt-debug-git-panel-content";
+    String REPOSITORIES_LIST_ID = "gwt-debug-git-panel-repositories";
+    String REPOSITORIES =
+        "//div[@id=\"" + REPOSITORIES_LIST_ID + "\"]/child::div[@id[starts-with(.,\"gwt-uid-\")]]";
+    String CHANGED_FILES_CONTAINER_ID = "gwt-debug-git-panel-changed-files";
+
+    String REPOSITORY_BY_NAME = REPOSITORIES + "//div[text()=\"%s\"]";
+    String CHANGES_LABEL_BY_REPOSITORY_NAME =
+        REPOSITORY_BY_NAME + "/parent::div/preceding-sibling::span[starts-with(text(),\"∓\")]";
+    String CHANGED_FILE_BY_NAME =
+        "//div[@id=\"" + CHANGED_FILES_CONTAINER_ID + "\"]/descendant::div[text()=\"%s\"]";
+  }
+
+  @FindBy(id = Locators.GIT_PANEL_BUTTON_ID)
+  private WebElement gitPanelButton;
+
+  @FindBy(id = Locators.GIT_PANEL_ID)
+  private WebElement gitPanel;
+
+  /** Opens git panel and waits until it will be ready. */
+  public void openPanel() {
+    uiWait.until(visibilityOf(gitPanelButton)).click();
+    uiWait.until(visibilityOf(gitPanel));
+  }
+
+  /** Returns number of repositories registered in the git panel. */
+  public int countRepositories() {
+    return seleniumWebDriver.findElements(By.xpath(Locators.REPOSITORIES)).size();
+  }
+
+  /**
+   * Waits until panel has specified number of git repositories.
+   *
+   * @param quantity expected number of repositories in the git panel
+   */
+  public void waitRepositories(int quantity) {
+    loadWait.until((ExpectedCondition<Boolean>) webDriver -> countRepositories() == quantity);
+  }
+
+  /** Returns true if given repository present into repositories list. */
+  public boolean isRepositoryPresent(String repository) {
+    return seleniumWebDriver
+            .findElements(By.xpath(String.format(Locators.REPOSITORY_BY_NAME, repository)))
+            .size()
+        == 1;
+  }
+
+  /**
+   * Waits until given repository presents in the repositories list.
+   *
+   * @param repository name of repository to wait for
+   */
+  public void waitRepositoryPresent(String repository) {
+    loadWait.until(
+        ExpectedConditions.presenceOfElementLocated(
+            By.xpath(String.format(Locators.REPOSITORY_BY_NAME, repository))));
```

Comment:
It's better to use static imports to improve readability.

Reply:
Well, I am just afraid that many static imports could make our code less clear.

Reply:
Believe me - in case of Page Object many static imports make code much clear. @tolusha can confirm it.

Reply:
I won't confirm that. It is matter of personal preferences and certain cases

Reply:
Hm. I thought you are supporter of usage of static imports, basing on your earlier comments like this  https://github.com/codenvy-legacy/qa/pull/644#discussion_r104435935 :-|
Anyway, usage of static import for constants like **Locators.REPOSITORY_BY_NAME** is obvious improvement so as **Locators** is [just an interface for constants](http://www.javapractices.com/topic/TopicAction.do?Id=32).